### PR TITLE
Compatibility with latest dask, pyarrow and deltalake

### LINKED ISSUE #65: Failed import when running `deltalake==0.14.0`
To reproduce:
- create new mamba/conda env
- `mamba install python==3.11 pip`
- `pip install deltalake==0.14`
- `pip install dask-deltatable`
- run `import dask_deltatable`

```
ImportError: cannot import name '_write_new_deltalake' from 'deltalake.writer' (/Users/rpelgrim/miniforge3/envs/deltalake-0130/lib/python3.11/site-packages/deltalake/writer.py)
```

To solve:
- `pip install deltalake==0.13`

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #65

cc @fjetter there is probably a better way to do this in case you all want to improve things here and make them more future-proof.

### CODE REVIEW

**Comment by fjetter** [comment]
I'll have a look and see what bothers CI

File: dask_deltatable/core.py [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 from functools import partial
-from typing import Any, cast
+from typing import Any, Callable, cast
 
+import dask
 import dask.dataframe as dd
 import pyarrow as pa
 import pyarrow.parquet as pq
 from dask.base import tokenize
+from dask.dataframe.io.parquet.arrow import ArrowDatasetEngine
```
Comment (fjetter) [inline dask_deltatable/core.py:13]: This import is a bit ugly but this class exposes our type mappers and this way we can align behavior

File: dask_deltatable/write.py [hunk 2]
```diff
@@ -216,9 +217,10 @@ def to_deltalake(
             configuration,
             storage_options,
             partition_filters,
+            custom_metadata,
         )
     }
-    graph = HighLevelGraph.from_collections(final_name, dsk, dependencies=(written,))
+    graph = HighLevelGraph.from_collections(final_name, dsk, dependencies=(written,))  # type: ignore
     result = Scalar(graph, final_name, "")
```
Comment (fjetter) [inline dask_deltatable/write.py:224]: I'll migrate to dask-expr in a follow up PR once this is green

**Comment by mrocklin** [comment]
CI is green.  @fjetter can I leave this PR with you?

**Comment by mrocklin** [comment]
Thanks Florian

**Pull request merged**